### ISSUE TITLE
How to understand negative partial distance correlation?

### ISSUE THREAD

**Issue by Yikai-Wang** [zhenxingjian/Partial_Distance_Correlation#6]
Thanks for your great work! 
I'm currently using PDC in my project and encountered an issue when attempting to calculate the partial distance correlation between two neural networks. I obtained a negative value, which seems abnormal according to Definition 3 in your paper. Can you provide some insight into how to interpret this result?

**Comment by zhenxingjian** [comment]
Hi Yikai,

Thanks for being interested in our work.

I've not met such issue during my training. But I may have some idea about negative PDC.
1. How many samples are you using to compute PDC? 
2. This might be a side-effect for the numerical stability, such as adding 1e-12 to make the gradient more stable. If PDC is very small negative value, this might be the issue.
3. What's the data dimension that you're using? If it's 1D feature (x\in \mathbb{R}), then you will need to remove the following:
        F.normalize()
    to make the feature meaningful. Otherwise, all features will be normalized to be const 1.

If the above ideas are not working, please let me know. It will be great if you can provide something that I can run so that I can have a better idea.

--
Xingjian

**Comment by Yikai-Wang** [comment]
My issue should fall under the second point. Thank you for your suggestion.
Again thanks for your great work!
